uniffi_bindgen/src/scaffolding.rs

```diff
@@ -61,27 +62,70 @@ mod filters {
             FFIType::RustBuffer => "uniffi::RustBuffer".into(),
             FFIType::RustError => "uniffi::deps::ffi_support::ExternError".into(),
             FFIType::ForeignBytes => "uniffi::ForeignBytes".into(),
+            FFIType::ForeignCallback => "uniffi::ForeignCallback".into(),
         })
     }
 
     pub fn lower_rs(nm: &dyn fmt::Display, type_: &Type) -> Result<String, askama::Error> {
         // By explicitly naming the type here, we help the rust compiler to type-check the user-provided
         // implementations of the functions that we're wrapping (and also to type-check our generated code).
-        Ok(format!(
-            "<{} as uniffi::ViaFfi>::lower({})",
-            type_rs(type_)?,
-            nm
-        ))
+        Ok(match type_ {
+            Type::CallbackInterface(type_name) => {
+                format!("<{}Proxy as uniffi::ViaFfi>::lower({})", type_name, nm,)
+            }
+            _ => format!("<{} as uniffi::ViaFfi>::lower({})", type_rs(type_)?, nm),
+        })
     }
 
     pub fn lift_rs(nm: &dyn fmt::Display, type_: &Type) -> Result<String, askama::Error> {
         // By explicitly naming the type here, we help the rust compiler to type-check the user-provided
         // implementations of the functions that we're wrapping (and also to type-check our generated code).
         // This will panic if the bindings provide an invalid value over the FFI.
-        Ok(format!(
-            "<{} as uniffi::ViaFfi>::try_lift({}).unwrap()",
-            type_rs(type_)?,
-            nm
-        ))
+        Ok(match type_ {
+            Type::CallbackInterface(type_name) => format!(
+                "Box::new(<{}Proxy as uniffi::ViaFfi>::try_lift({}).unwrap())",
+                type_name, nm,
+            ),
+            _ => format!(
+                "<{} as uniffi::ViaFfi>::try_lift({}).unwrap()",
+                type_rs(type_)?,
+                nm
+            ),
+        })
+    }
+
+    /// Get a Rust expression for writing a value into a byte buffer.
+    pub fn write_rs(
+        nm: &dyn fmt::Display,
+        target: &dyn fmt::Display,
+        type_: &Type,
+    ) -> Result<String, askama::Error> {
+        Ok(match type_ {
+            Type::CallbackInterface(type_name) => format!(
+                "Box::new(<{}Proxy as uniffi::ViaFfi>::write(&{}, {})",
+                type_name, nm, target,
```

Comment:
I don't think it should be possible to write `CallbackInterfaces` into a buffer, and it may be better to bail out here rather than generate code for it at all. (Also I think the closing paren on `Box::new(` may be missing here?)